winml/adapter/winml_adapter_dml.cpp

```diff
@@ -22,30 +22,25 @@ namespace winmla = Windows::AI::MachineLearning::Adapter;
 
 EXTERN_C IMAGE_DOS_HEADER __ImageBase;
 
-static std::string CurrentModulePath() {
-  char path[MAX_PATH];
-  FAIL_FAST_IF(0 == GetModuleFileNameA((HINSTANCE)&__ImageBase, path, _countof(path)));
+static std::wstring CurrentModulePath() {
+    WCHAR path[MAX_PATH];
+    FAIL_FAST_IF(0 == GetModuleFileNameW((HINSTANCE)&__ImageBase, path, _countof(path)));
 
-  char absolute_path[MAX_PATH];
-  char* name;
-  FAIL_FAST_IF(0 == GetFullPathNameA(path, _countof(path), absolute_path, &name));
+    WCHAR absolute_path[MAX_PATH];
+    WCHAR* name;
+    FAIL_FAST_IF(0 == GetFullPathNameW(path, _countof(path), absolute_path, &name));
 
-  auto idx = std::distance(absolute_path, name);
-  auto out_path = std::string(absolute_path);
-  out_path.resize(idx);
+    auto idx = std::distance(absolute_path, name);
+    auto out_path = std::wstring(absolute_path);
```

Comment:
I think `auto out_path = std::wstring(absolute_path, idx);` does one copy with the right size